### PR TITLE
Support podman-remote

### PR DESCRIPTION
**Pull request by michaellevy101**
This PR supports podman-remote for test-infra, now the images will be visible to the host container tool.

### CODE REVIEW

File: scripts/run_minikube.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ function init_minikube() {
     for i in {1..5}
     do
         minikube delete
-        minikube start --driver=kvm2 --memory="${MINIKUBE_RAM_MB}" --cpus=4 --force --wait-timeout=15m0s --disk-size="${MINIKUBE_DISK_SIZE}" --addons=registry || true
+        minikube start --driver=kvm2 --listen-address=0.0.0.0 --memory="${MINIKUBE_RAM_MB}" --cpus=4 --force --wait-timeout=15m0s --disk-size="${MINIKUBE_DISK_SIZE}" --addons=registry || true
```
Comment (adriengentil) [inline scripts/run_minikube.sh:29]:
```
    --listen-address='':
        IP Address to use to expose ports (docker and podman driver only)
```
should it be cleaned up?

File: scripts/utils.sh [hunk 2]
```diff
@@ -185,4 +185,49 @@ function configure_none_platform_iptables_rules() {
...
 }
 
+function running_from_skipper() {
+   # The SKIPPER_UID environment variable is an indication that we are running on a skipper container.
+   [ -n "${SKIPPER_UID+x}" ]
+}
+
+function get_container_runtime_command() {
+  # if CONTAINER_TOOL is defined skipping
+  if [ ! -n "${CONTAINER_TOOL+x}" ]; then
```
Comment (adriengentil) [inline scripts/utils.sh:195]:
```suggestion
  if [ -z "${CONTAINER_TOOL+x}" ]; then
```
looks clearer

File: scripts/utils.sh [hunk 3]
```diff
@@ -185,4 +185,49 @@ function configure_none_platform_iptables_rules() {
...
+    if running_from_skipper; then
+      if [ -z ${CONTAINER_RUNTIME_COMMAND} ]; then
+        echo "CONTAINER_RUNTIME_COMMAND doesn't set on old skipper version -> default to docker. Upgrade your skipper to the latest version"
+      fi
+
+      if [ "${CONTAINER_RUNTIME_COMMAND:-docker}" == "docker" ]; then
```
Comment (adriengentil) [inline scripts/utils.sh:201]:
```suggestion
      if [ "${CONTAINER_RUNTIME_COMMAND:-docker}" = "docker" ]; then
```

File: scripts/install_environment.sh [hunk 4]
```diff
@@ -146,15 +147,25 @@ EOF
     chmod +x "${hook_filename}"
 }
 
+function install_podman(){
```
Comment (adriengentil) [inline scripts/install_environment.sh:150]: nit: for consistency, use the same indentation level as in the rest of the file

File: scripts/utils.sh [hunk 5]
```diff
@@ -185,4 +185,49 @@ function configure_none_platform_iptables_rules() {
...
+    exit
+  fi
+
+  if [ "$(get_container_runtime_command)" = "podman-remote" ]; then
+    if podman-remote4 info 2>&1 | grep "server API version is too old" &> /dev/null; then
+      echo "using podman-remote version 3"
+      ln $(which podman-remote3) /tools/podman-remote
+    else
+      echo "using podman-remote version 4"
+      ln $(which podman-remote4) /tools/podman-remote
```
Comment (adriengentil) [inline scripts/utils.sh:228]: what is `/tools` directory? it is part of skipper magic?

**Review by adriengentil — CHANGES_REQUESTED** [review]

**Comment by michaellevy101** [comment]
@adriengentil It's not ready yet

File: scripts/run_minikube.sh (hunk 1 above)
Comment (michaellevy101) [reply scripts/run_minikube.sh:29]: yes, This PR isn't ready yet

File: scripts/utils.sh (hunk 5 above)
Comment (michaellevy101) [reply scripts/utils.sh:228]: no, should be create as part of the image

File: scripts/install_environment.sh [hunk 6]
```diff
@@ -146,20 +147,40 @@ EOF
...
-    if ! [ -x "$(command -v docker)" ] && ! [ -x "$(command -v podman)" ]; then
-        sudo dnf install podman -y
-    elif [ -x "$(command -v podman)" ]; then
-        current_version="$(head -n1 <(podman version) | awk '{print $2}')"
-        minimum_version="1.6.4"
-        if ! version_is_greater "$current_version" "$minimum_version"; then
-            sudo dnf install podman-$minimum_version -y
-        fi
+    if [ -x "$(command -v docker)" ]; then
+      echo "docker is already installed"
```
Comment (osherdp) [inline scripts/install_environment.sh:163]: fix indentation to be 4 spaces like the rest of this file

File: scripts/install_environment.sh [hunk 7]
```diff
@@ -146,20 +147,40 @@ EOF
...
+      return
+    fi
+
+    if [ ! -x "$(command -v podman)" ]; then
+      install_podman
```
Comment (osherdp) [inline scripts/install_environment.sh:168]: if on some environment (and I think it's most local environments) we have podman installed but without the podman socket, then the podman socket will not be configured
maybe we should check if there's an active socket and based on that do or don't do the socket configuration

File: skipper.yaml [hunk 8]
```diff
@@ -36,8 +36,14 @@ volumes:
   - /dev/:/dev
   - /run/udev:/run/udev
 
+  # podman - sharing the podman.socket between the host and the skipper container
+  - $XDG_RUNTIME_DIR/podman/podman.sock:/run/podman/podman.sock
```
Comment (osherdp) [inline skipper.yaml:40]: in assisted-service we have ``/run/podman/podman.sock:/run/podman/podman.sock``. what is it different?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/consts/consts.py [hunk 9]
```diff
@@ -125,7 +125,6 @@ def all(cls):
 WORKER_TF_CPU_MODE = HOST_PASSTHROUGH_CPU_MODE
 NUMBER_OF_MASTERS = 3
 NAMESPACE_POOL_SIZE = 15
-PODMAN_FLAGS = "--cgroup-manager=cgroupfs --storage-driver=vfs --events-backend=file"
```
Comment (eliorerz) [inline src/consts/consts.py:128]: I just saw `--cgroup-manager=cgroupfs --storage-driver=vfs --events-backend=file` is hard-coded in https://github.com/openshift/assisted-test-infra/blob/c5d5a2867a6f552d250e39e0b3513c8fdfd33750/src/assisted_test_infra/test_infra/controllers/tang_controller/tang_controller.py#L38

File: skipper.yaml (hunk 8 above)
Comment (michaellevy101) [reply skipper.yaml:40]: It depends on the podman-socket installation.

**Review by eliorerz — APPROVED** [review]
LGTM
/hold

**Pull request merged**